tests/solidity_test_helpers/ZeroX/ZeroXPoC.sol

```diff
@@ -0,0 +1,446 @@
+pragma solidity 0.4.20;
+
+import 'reporting/IMarket.sol';
+import 'trading/IShareToken.sol';
+import 'trading/ICash.sol';
+import 'libraries/token/ERC20.sol';
+import 'libraries/math/SafeMathUint256.sol';
+import 'libraries/ReentrancyGuard.sol';
+import 'trading/CompleteSets.sol';
+import 'Augur.sol';
+import 'IController.sol';
+
+contract ZeroXPoC is ReentrancyGuard {
+    using SafeMathUint256 for uint256;
+
+    // Error Codes
+    enum Errors {
+        ORDER_EXPIRED,                    // Order has already expired
+        ORDER_FULLY_FILLED_OR_CANCELLED,  // Order has already been fully filled or cancelled
+        ROUNDING_ERROR_TOO_LARGE,         // Rounding error too large
+        INSUFFICIENT_BALANCE_OR_ALLOWANCE // Insufficient balance or allowance for token transfer
+    }
+
+    uint16 constant public EXTERNAL_QUERY_GAS_LIMIT = 4999;    // Changes to state require at least 5000 gas
+
+    // token => account => balance
+    mapping(address => mapping(address => uint256)) public tokenBalances;
+
+    // orderHash => amounts of Amount filled or cancelled.
+    mapping (bytes32 => uint) public filled;
+    mapping (bytes32 => uint) public cancelled;
+
+    Augur public augur;
+    CompleteSets public completeSets;
+    IController public controller;
+    ICash public cash;
+
+    event Fill(
+        address taker,
+        uint256 amountFilled,
+        bytes32 orderHash
+    );
+
+    event Cancel(
+        bytes32 orderHash,
+        uint256 cancelledAmount
+    );
+
+    event Error(
+        uint8 indexed errorId,
+        bytes32 indexed orderHash
+    );
+
+    struct Order {
+        address maker;
+        IMarket market;
+        uint256 outcome;
+        uint256 orderType;
+        uint256 amount;
+        uint256 price;
+        uint expirationTimestampInSec;
+        bytes32 orderHash;
+    }
+
+    event Deposit(
+        address indexed account,
+        address indexed shareToken,
+        uint256 amountDeposited,
+        uint256 amountHeld
+    );
+
+    event Withdraw(
+        address indexed account,
+        address indexed shareToken,
+        uint256 amountWithdrawn,
+        uint256 amountHeld
+    );
+
+    function ZeroXPoC(Augur _augur) public {
+        augur = _augur;
+        controller = _augur.getController();
+        completeSets = CompleteSets(controller.lookup("CompleteSets"));
+        cash = ICash(controller.lookup("Cash"));
+        cash.approve(augur, 2**254);
```

Comment:
`2 ** 256 - 1` saves gas

Reply:
Yea I noted above there is no attempt made here to save on gas or be efficient (or even be well coded) since its just to confirm this is possible post-deploy